packages/api-explorer/__tests__/SchemaTabs.test.jsx

```diff
@@ -41,31 +49,85 @@ const props = {
 
+jest.mock('json-schema-faker')
+
 describe('SchemaTabs', () => {
   let element
   let block
 
   beforeEach(() => {
-    element = shallow(
-      <SchemaTabs
-        {...props}
-      />
-    )
+    element = shallow(<SchemaTabs{...props} />)
     block = element.find(BlockWithTab)
   })
 
-  test('renders a tab with 2 items', () => {
+  test('renders a tab with 3 items', () => {
     expect(block).toHaveLength(1)
     const { items } = block.props()
+    const exampleLabel = <FormattedMessage defaultMessage="Example" id="schemaTabs.label.example" />
     const requestLabel = <FormattedMessage defaultMessage="Request" id="schemaTabs.label.request" />
     const responseLabel = <FormattedMessage defaultMessage="Response" id="schemaTabs.label.response" />
-    expect(items).toEqual([{ value: 'request', label: requestLabel }, { value: 'response', label: responseLabel }])
+    expect(items).toEqual([
+      { value: 'example', label: exampleLabel },
+      { value: 'request', label: requestLabel },
+      { value: 'response', label: responseLabel }
+    ])
   })
 
   test('set correct tab when one is clicked', () => {
-    block.prop('onClick')('request')
-    element.update()
-    expect(block.prop('selected')).toEqual('request')
+    expect(element.find(BlockWithTab).prop('selected')).toEqual('example')
+    element.find(BlockWithTab).simulate('click', 'response')
+    expect(element.find(BlockWithTab).prop('selected')).toEqual('response')
+  })
+
+  describe('if example is selected', () => {
+    beforeEach(() => {
+      jest.clearAllMocks()
+    })
+
+    test('and render example if requestBody.example is set', (done) => {
+      element = shallow(<SchemaTabs oas={oasWithExample} operation={operationWithExample} />)
+      const petType = 'Carlino'
+      const petChildren = ['john', 'doo']
+      setTimeout(() => {
+        element.update()
+        expect(element.find(JsonViewer).prop('schema')).toEqual({pet_type: petType, pet_children: petChildren})
+        done()
+      }, 0)
+    })
+
+    test('render generated example when requestBody.example is not set', (done) => {
+      const petId = 1234
+      // eslint-disable-next-line no-underscore-dangle
+      jsf._generateReturnValue(() => ({petId}))
+
+      element = shallow(<SchemaTabs {...props} />)
+      setTimeout(() => {
+        element.update()
+        expect(element.find(JsonViewer).prop('schema')).toEqual({petId})
+        done()
+      }, 0)
+
+    })
+
+    test('render missing schema message', () => {
+      element = shallow(<SchemaTabs oas={{}} operation={{}} />)
+      const formattedMessage = element.find(FormattedMessage)
+      expect(formattedMessage).toHaveLength(1)
+      expect(formattedMessage.prop('id')).toEqual('schemaTabs.missing.example')
+    })
+
+    test('render missing schema alert ', (done) => {
+      // eslint-disable-next-line no-underscore-dangle
+      jsf._generateReturnValue(() => {
+        throw new Error('Maximum call stack size exceeded')
```

Comment:
This is an error of missing schema alert?

Reply:
no, it's an error in the test title